addons/osfstorage/tests/test_views.py

```diff
@@ -1411,6 +1415,44 @@ def test_file_remove_tag_fail_doesnt_create_log(self, mock_log):
 @pytest.mark.enable_bookmark_creation
 class TestFileViews(StorageTestCase):
 
+    def add_github(self):
+        addon = self.node.add_addon('github', auth=Auth(self.user))
+        oauth_settings = GitHubAccountFactory()
+        oauth_settings.save()
+        self.user.add_addon('github')
+        self.user.external_accounts.add(oauth_settings)
+        self.user.save()
+        addon.user_settings = self.user.get_addon('github')
+        addon.external_account = oauth_settings
+        addon.repo = 'something'
+        addon.user = 'someone'
+        addon.save()
+        addon.user_settings.oauth_grants[self.project._id] = {
+            oauth_settings._id: []}
+        addon.user_settings.save()
+        self.node.save()
+
+
+    @responses.activate
+    def test_file_view_updates_history(self):
+        self.add_github()
```

Comment:
nit: assert that file.history doesn't exist at start of test to demonstrate what changed

Reply:
This is hard to test for because the filenode doesn't exist at all at the beginning of the test. As I wrote in my comment the files represent addon files that were added via that addon only. So if you dropped a file in Dropbox it has to create that filenode on the OSF-side and then update the `history`, that's why I can't check the `history`, figurative the file is still sitting un-viewed in Dropbox at that point not in our DB.

Reply:
So the WB callback is what creates the filenode in the OSF database? Just trying to figure out how we can make this test a good reference for newcomers to understand how things work, but if it's just too convoluted, I guess it's just too convoluted and that's a separate problem.

Reply:
> So the WB callback is what creates the filenode in the OSF database? 

No, the filenode is created as the file is seen via the view, [here](https://github.com/CenterForOpenScience/osf.io/blob/develop/addons/base/views.py#L729) the history is added to via the `touch` function. The WB callback is used for a bunch of metrics, but not this type.